client/src/views/Login/Login.js

```diff
@@ -0,0 +1,253 @@
+          fullWidth
+          inputRef={register({
+            required: 'Email address is compulsory',
+            pattern: {
+              value: EmailRegex,
+              message: 'Invalid email address',
+            },
+          })}
+          error={Boolean(errors.email)}
+          helperText={errors.email && errors.email.message}
+        />
+        <StyledTextField
+          name="password"
+          label="Password"
+          variant="outlined"
+          placeholder="Enter Password here"
+          fullWidth
+          type={showPassword ? 'text' : 'password'}
+          inputRef={register({ required: 'Password field is compulsory' })}
+          error={Boolean(errors.password)}
+          helperText={errors.password && errors.password.message}
+          InputProps={{
+            endAdornment: (
+              <InputAdornment position="end">
+                <IconButton
+                  className={style.icon}
+                  aria-label="toggle password visibility"
+                  onClick={handleClickShowPassword}
+                >
+                  {showPassword ? <Visibility /> : <VisibilityOff />}
+                </IconButton>
+              </InputAdornment>
+            ),
+          }}
+          onChange={() => {
+            triggerValidation('confirmPassword');
+          }}
+        />
+        <StyledTextField
+          name="confirmPassword"
+          label="Confirm Password"
+          variant="outlined"
+          placeholder="Retype your password"
+          fullWidth
+          type="password"
+          inputRef={register({
+            required: 'Retype your password here',
+            validate: {
+              match: (value) =>
+                value === watchPassword ? true : 'Passwords are not matching',
+              message: 'Invalid Password',
+            },
+          })}
+          error={Boolean(errors.confirmPassword)}
+          helperText={errors.confirmPassword && errors.confirmPassword.message}
+        />
+        <StyledButton type="submit" fullWidth>
+          <Typography variant="button">Signup</Typography>
+        </StyledButton>
+      </form>
+      <HyperLink onClick={toggleView}>
+        <Typography variant="subtitle2" align="center">
+          Already have an account ? Login
+        </Typography>
+      </HyperLink>
+    </React.Fragment>
+  );
+};
+
+const Login = (props) => {
+  const [isRegistered, setisRegistered] = useState(true);
+  const toggleView = () => setisRegistered(!isRegistered);
+
+  return (
+    <section className={style.rootContainer}>
+      <a href="/" className={style.logoWrapper}>
+        <img
+          className={style.logo}
+          src={process.env.PUBLIC_URL + '/logo.png'}
+          alt="Website logo"
+        />
+      </a>
+      <div className={style.mainView}>
+        {isRegistered ? (
+          <LoginView toggleView={toggleView} />
+        ) : (
+          <SignUpView toggleView={toggleView} />
```

Comment:
IMO, there is only a text field diff in the sign up and the login view, you can merge these views ,instead of creating two seperate views.

Reply:
Ok, but there is one more field to signup for user role in signup and add routing logic based on user role. I will create a new PR for that.